prow/jira/fakejira/fake.go

```diff
@@ -116,44 +131,73 @@ func (f *FakeClient) CreateIssueLink(link *jira.IssueLink) error {
 	if err != nil {
 		return fmt.Errorf("failed to get outward link issue: %v", err)
 	}
-	outward.Fields.IssueLinks = append(outward.Fields.IssueLinks, link)
+	// links in an issue struct do not include the short definition of the the issue they are part of.
+	// This behavior is used by the external jira-lifecycle hook plugin to identify which direction a clone happened,
```

Comment:
I also don't think comments about downstream consumers of the library are a good idea here. Regardless of who uses this, the link is a valid thing to set, right?